plugins/de.cognicrypt.staticanalyzer.kotlin/src/de/cognicrypt/staticanalyzer/kotlin/utilities/KotlinUtils.java

```diff
@@ -0,0 +1,261 @@
+package de.cognicrypt.staticanalyzer.kotlin.utilities;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.maven.model.Dependency;
+import org.apache.maven.model.Model;
+import org.apache.maven.model.io.xpp3.MavenXpp3Reader;
+import org.apache.maven.model.io.xpp3.MavenXpp3Writer;
+import org.apache.maven.project.MavenProject;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.resources.IncrementalProjectBuilder;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.OperationCanceledException;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.IPackageFragment;
+import org.eclipse.jdt.core.IPackageFragmentRoot;
+import org.eclipse.jdt.core.JavaCore;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.widgets.MessageBox;
+import org.eclipse.ui.IWorkbenchWindow;
+import org.eclipse.ui.PlatformUI;
+import org.jetbrains.kotlin.core.compiler.KotlinCompiler;
+import org.jetbrains.kotlin.core.compiler.KotlinCompilerResult;
+import org.jetbrains.kotlin.core.model.KotlinNature;
+
+import de.cognicrypt.staticanalyzer.kotlin.Activator;
+
+public class KotlinUtils {
+	
+	public static void compileKotlinFiles(IProject ip) {
+		IJavaProject javaProject = JavaCore.create(ip);
+		
+		if(Platform.getBundle("org.jetbrains.kotlin.core") != null) {
+ 			if(KotlinNature.hasKotlinNature(ip)) {
```

Comment:
If the conditions in l48 and l49 need to both be true for anything to happen, just conjunct them in one if clause.